src/SubscribeRequest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpMqtt\Client;
+
+use DateTime;
+
+/**
+ * Represents an unsubscribe request.
+ *
+ * @package PhpMqtt\Client
+ */
+class SubscribeRequest extends PendingMessage
+{
+    /** @var array<string> */
+    private $topicFilters;
+
+    /** @var ... */
+    private $qualitiesOfService;
+
+    /**
+     * Creates a new unsubscribe request object.
+     *
+     * @param int           $messageId
+     * @param string        $topicFilters
+     * @param DateTime|null $sentAt
+     */
+    public function __construct(int $messageId, array $topicFilters, array $qualitiesOfService)
```

Comment:
Decoupling QoS from the topic feels a bit weird. Not sure introducing another DTO is worth it, but we could at least use an array-tuple: `[$topic, $qos]`